programs/ssl/ssl_server2.c

```diff
@@ -3643,6 +3646,12 @@ int main( int argc, char *argv[] )
     }
 #endif
 
+#if defined(MBEDTLS_MEMORY_DEBUG)
+    mbedtls_memory_buffer_alloc_max_get(&max_mem_used, &max_blocks);
+    mbedtls_memory_buffer_alloc_max_reset();
+    mbedtls_printf( " Memory peak usage after handshake: %zd bytes\n", max_mem_used);
```

Comment:
It looks like the handshake is completely complete right before `7c. Continue doing data exchanges?`. We should print out the current memory usage right before `7c` (not max usage), using `mbedtls_memory_buffer_alloc_cur_get()`.